core-bundle/src/Resources/contao/library/Contao/Dbafs.php

```diff
@@ -494,9 +494,17 @@ public static function syncFiles()
 		}
 
 		$objDatabase = Database::getInstance();
+		$objDatabaseEngine = $objDatabase
```

Comment:
Can't we use Doctrine DBAL instead and use `->beginTransaction()`? :)

Reply:
Of course, but what's the point if not rewriting the old code around it that uses the old connection? :-)

Reply:
Yeah. The `Dbafs` would be such a wonderful example for a service with tests etc... 😄

Reply:
May be @m-vo should completely re-implement is as a service? 😁

Reply:
Maybe that should then be another PR to enhance chances of it ever getting merged. 😁 But yeah, totally agree with both of you...